framework/src/modules/interoperability/sidechain/commands/message_recovery.ts

```diff
@@ -0,0 +1,69 @@
+import { codec } from '@liskhq/lisk-codec';
+import { CommandExecuteContext, VerificationResult } from '../../../../node/state_machine';
+import { BaseInteroperabilityCommand } from '../../base_interoperability_command';
+import { COMMAND_ID_MESSAGE_RECOVERY, EMPTY_FEE_ADDRESS } from '../../constants';
+import { createCCCommandExecuteContext } from '../../context';
+import { ccmSchema, messageRecoveryParams } from '../../schema';
+import { CCMsg, StoreCallback, MessageRecoveryParams } from '../../types';
+import { swapChainIDs } from '../../utils';
+import { SidechainInteroperabilityStore } from '../store';
+
+export class MessageRecoveryCommand extends BaseInteroperabilityCommand {
+	public id = COMMAND_ID_MESSAGE_RECOVERY;
+	public name = 'messageRecovery';
+	public schema = messageRecoveryParams;
+
+	// TODO
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async verify(): Promise<VerificationResult> {
+		throw new Error('Method not implemented.');
+	}
+
+	public async execute(context: CommandExecuteContext<MessageRecoveryParams>): Promise<void> {
```

Comment:
there is some logic missing in this sidechain `execute` that is present in the mainchain `execute`?

Reply:
I thought updating `terminatedOutboxAccount` happens only in mainchain due to comment `# Update sidechain outbox root` in the pseudocode of the LIP 54. Isn't this the case?

Reply:
I believe we may need that logic for direct channels implementation. @ricott1  can you confirm?

Reply:
We do need this logic in general, because after we have direct sidechain channels you would have to maintain the outbox root of the terminated sidechain also on a sidechain.